x-pack/filebeat/tests/integration/framework_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type BeatProc struct {
+	Binary  string
+	Args    []string
+	Cmd     *exec.Cmd
+	t       *testing.T
+	tempDir string
+}
+
+// NewBeat createa a new Beat process from the system tests binary.
+// It sets some requried options like the home path, logging, etc.
+func NewBeat(t *testing.T, binary string, args []string, tempDir string) BeatProc {
```

Comment:
[NIT]

YOu could have `args` as a variadic argument

```suggestion
func NewBeat(t *testing.T, binary string, tempDir string, args ...string) BeatProc {
```